assets/src/edit-story/components/canvas/displayLayer.js

```diff
@@ -42,13 +42,13 @@ const DisplayPageArea = styled(PageArea).attrs({
 `;
 
 function DisplayLayer() {
-  const {
-    state: { currentPage },
-  } = useStory();
-  const {
-    state: { editingElement },
-    actions: { setPageContainer },
-  } = useCanvas();
+  const { currentPage } = useStory((state) => ({
+    currentPage: state.state.currentPage,
+  }));
```

Comment:
I'm adding this as a comment here, even though it doesn't apply to this selector in particular:

Would it make sense to over time develop more specific hooks for common selections are shortcuts?

So this one would be just:

```js
  const { currentPage } = useCurrentPage();
```

But other occasions might be a little less specific like:

```js
  const { updateElementsByResourceId } = useStoryAction('updateElementsByResourceId');
```

Also, is there any reason that we return an object with variables as keys here? Is that required by the underlying `useSelector` helper?

I.e. why couldn't this original code here not just be:

```js
  const currentPage = useStory((state) => state.state.currentPage);
```

And thus the suggested helpers be:

```js
  const currentPage = useCurrentPage();
  const updateElementsByResourceId = useStoryAction('updateElementsByResourceId');
```

Reply:
> const { currentPage } = useCurrentPage();

I'm not sure about this one, as it obfuscates the context the attribute is queried from, and require a lot of them (eg. usePages, useIsSaving + 10 others).

> const { updateElementsByResourceId } = useStoryAction('updateElementsByResourceId');

This may be more useful, it's easy to misspell the attribute name in the string but it's also easy to pick that up in code review (and, javascript). This version is more type safe if typescript were introduced later:
> const currentPage = useStory((state) => state.state.currentPage);

I can see a couple of downsides to this:

1. If more than one attribute is needed, it's more work to refactor the call to useXX to add the additional state or actions, without making multiple calls to 'useContextSelector' which is probably unnecessary. For example, refactoring this:

const currentPage = useStory((state) => state.state.currentPage);

to

const { selectedElements, stylePresets, updateStory } = useStory(
  ({
    state: { selectedElements, story: { stylePresets } },
    actions: { updateStory },
  }) => ({ selectedElements, stylePresets, updateStory })
);

Most developers will be lazy and do this instead:

const currentPage = useStory((state) => state.state.currentPage);
const stylePresets = useStory((state) => state.state.story.stylePresets);
const updateStory = useStory((state) => state.actions.updateStory);

which means 3 calls to useContextSelector.

2. This isn't so important, but useStory() performs a shallow comparison selector. This means all of the attributes of 'currentPage' are compared instead of only the reference to currentPage, which may be unnecessary. Of the order of microseconds not ms, so this may not matter very much.

3. Some state/actions have more nesting, like 'state.story.stylePresets'.

In a separate PR, if everyone likes this approach I can change the above useStory() example (as an example) to this:

const currentPage = useStory('currentPage');
const stylePresets = useStory((state) => state.state.story.stylePresets);
const updateStory = useStoryAction('updateStory');

This disregards caveats 2 and 3 above :D but perhaps this is a readability improvement and is worth it.

WDYT?

Reply:
Let's work on the API in a separate PR. Several of your points are excellent and we need to strike the right balance between UX and DX.